### DISCUSSION TITLE
Redundant Equality CodeFix for "... == false"

### DISCUSSION THREAD

**Discussion by awaescher** [JosefPihrt/Roslynator#860] [Ideas]
Visual Studio has a built-in CodeFix for "Redundant Equality" already but only if the comparison is done against `True`

![image](https://user-images.githubusercontent.com/3630638/141143746-a22dabc3-a572-47e0-ab66-a9d8d7688ec1.png)

I would really love to have a CodeFix to convert `... = False` to `Not ...`.

While I know comparing against `False` is not wrong technically, I think it's pretty old fashioned to use.

### C#
```csharp
HttpResponseMessage response;

// bad
if (response.IsSuccessStatusCode == false)
    throw ...;

// good
if (!response.IsSuccessStatusCode)
    throw ...;
```

### VB.NET
```basic
Dim response As HttpResponseMessage

' bad
If response.IsSuccessStatusCode = False Then
    Throw ...
End If

' good
If Not response.IsSuccessStatusCode Then
    Throw ...
End If
```

**Comment by josefpihrt** [comment]
Hi, this analyzer is already part of Roslynator: https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1049.md

**Reply by awaescher** [reply]
Whoa, how did I miss that? I tried to get a CodeFix in Visual Studio quite a few times now.
Sorry for burning your valuable time.

As I am looking to use this on a huge VB code base ... could it be that is for C# only?

**Comment by josefpihrt** [comment]
Unfortunately yes, almost all Roslynator analyzers are for C# only.

**Reply by awaescher** [reply]
Yeah, that's what I thought. Totally understandable but a pity for people maintaining legacy VB code 😉
Nevermind, I am really grateful for the work you are doing, Roslynator is a great extension. Thank you!
